stdlib/enum.pyi

```diff
@@ -148,6 +150,18 @@ class EnumMeta(type):
             type: type | None = None,
             start: int = 1,
         ) -> type[Enum]: ...
+
+    # Overload 3 (py312+ only): Value lookup on an already existing enum class (complex case)
+    #
+    # >>> class Foo(enum.Enum):
+    # ...     X = 1, 2, 3
+    # >>> Foo(1, 2, 3)
+    # <Foo.X: (1, 2, 3)>
+    #
+    if sys.version_info >= (3, 12):
+        @overload
+        def __call__(cls: type[_EnumMemberT], value: Any, *values: Any) -> _EnumMemberT: ...
```

Comment:
I added a third overload here rather than modifying the first overload, as if this were the first overload, there'd likely be cases where type checkers _should_ be selecting the functional-API overload but would go for this one instead. It's a very lenient signature.